bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/server/enable/HasLanguageServerPropertyTester.java

```diff
@@ -38,9 +48,15 @@ public boolean test(Object receiver, String property, Object[] args, Object expe
 			if (receiver instanceof URI) {
 				// called from the language server enabler for LSP4E:
 				var uri = (URI) receiver;
+				if (invalidContentType(uri))
+					return false;
 				// when getProject is empty, it's an external file: Check if the file is already opened, if not check the active editor:
-				return getProject(uri).map(cLanguageServerProvider::isEnabledFor)
+				var isEnabled = LspUtils.getProject(uri).map(cLanguageServerProvider::isEnabledFor)
 						.orElse(LspUtils.isFileOpenedInLspEditor(uri));
+				if (isEnabled) {
+					initialFileManager.setInitialUri(uri);
```

Comment:
I agree. This is a bit hacky, but this is the only point were LSP4E tells us which file/uri he wants to open. I think about a PR to establish an interface in LSP4E for this.

Reply:
It would be ideal when the rootUri argument in  the `getInitializationOptions` method form LSP4E is the URi to the file to be opened. Sadly it only the parent if the file. So we lost this information.